UPGRADE.md

```diff
@@ -1,5 +1,103 @@
 # Upgrade to 2.7
 
+## Deprecated code generators and related console commands
+ 
+These console commands have been deprecated:
+
+ * `orm:convert-mapping`
+ * `orm:generate:entities`
+ * `orm:generate-repositories`
+
+These classes have been deprecated:
+
+ * `Doctrine\ORM\Tools\EntityGenerator`
+ * `Doctrine\ORM\Tools\EntityRepositoryGenerator`
+
+Whole Doctrine\ORM\Tools\Export namespace with all its members have been deprecated as well.
+
+## Deprecated `Doctrine\ORM\Proxy\Proxy` marker interface
+
+Proxy objects in Doctrine 3.0 will no longer implement `Doctrine\ORM\Proxy\Proxy` nor
+`Doctrine\Common\Persistence\Proxy`: instead, they implement
+`ProxyManager\Proxy\GhostObjectInterface`.
+
+These related classes have been deprecated:
+
+ * `Doctrine\ORM\Proxy\ProxyFactory`
+ * `Doctrine\ORM\Proxy\Autoloader` - we suggest using the composer autoloader instead
```

Comment:
using the composer autoloader for generated proxy classes ? This won't work well. One of the point of the `Doctrine\ORM\Proxy\Autoloader` is that it would generate the proxy class if needed (allowing to unserialize your entities even in case the proxies are not generated yet, which is very handy during dev where you probably generate the proxies on demand rather than dumping them).

Reply:
I disagree, you can just use `eval()` in dev.

Reply:
The strategy being used is totally unrelated to that. The `Doctrine\ORM\Proxy\Autoloader` is about calling the proxy generator, not about changing how the proxy class is defined.

When the ORM wants to instantiate a proxy class, it will take care of generating it first. The autoloader is solving the case where you call `unserialize()`, and the serialized entity was actually an initialized proxy (and so the proxy class name appears in the serialized string, not the entity class name).